docs/proposals/2432-gpu-testing-on-llm-blueprints/README.md

```diff
@@ -0,0 +1,180 @@
+# KEP-2432: GPU Testing for LLM Blueprints
+
+Project Page:[ GPU Testing for LLM Blueprints](https://www.kubeflow.org/events/gsoc-2025/#project-7-gpu-testing-for-llm-blueprints)
+
+Issue:[ https://github.com/kubeflow/trainer/issues/2432](https://github.com/kubeflow/trainer/issues/2432)
+
+Mentors:[ @andreyvelich](https://github.com/andreyvelich),[ @varodrig](https://github.com/varodrig)
+
+Project Size: 350 hrs
+
+
+## Summary
+
+This project aims to use self-hosted runners to run GPU-intensive tasks like LLM blueprint or (planned) AI Playground. The necessary infra is provided by Oracle, plan is to use Oracle Kubernetes Engine (OKE) with NVIDIA GPUs for this task. Any code or sample that requires GPU-intensive resources will be transferred to OKE infra instead of generic GitHub infra for faster and more efficient execution.
+
+To ensure robust and efficient testing, a dedicated policy will be implemented: whenever code is updated in either of the following notebooks—
+
+- DeepSpeed: `master/examples/deepspeed/text-summarization/T5-Fine-Tuning.ipynb`
+- TorchTune: `master/examples/torchtune/llama3_2/alpaca-trainjob-yaml.ipynb`
+
+The workflow will monitor changes in these paths and, upon approval from maintainer, execute the notebooks using a GitHub self-hosted runner on the OKE infrastructure. The GitHub Action will be triggered only when a specific label (e.g., `run-gpu-e2e`) is applied to a pull request. For security, this workflow requires the lavel to be added from a maintainer before execution.
```

Comment:
@jaiakash Please can you update the KEP based on our recent conversation that we always run this action, but it will just succeed if appropriate label doesn't exist in the PR.

Reply:
Done